src/sidebar/helpers/annotation-segment.ts

```diff
@@ -29,3 +30,18 @@ export function annotationMatchesSegment(
         stripCFIAssertions(selector.cfi) === stripCFIAssertions(segment.cfi)),
   );
 }
+
+/**
+ * Return true if the document segment displayed in a guest frame matches the
+ * configured focus filters.
+ */
+export function segmentMatchesFocusFilters(
+  segment: SegmentInfo,
+  filters: Filters,
+): boolean {
+  if (segment.cfi && filters.cfi) {
+    const [cfiStart, cfiEnd] = filters.cfi.value.split('-');
+    return cfiInRange(segment.cfi, cfiStart, cfiEnd);
+  }
+  return true;
```

Comment:
In this particular case, the `if` block is pretty small, so it would not be critical, but I would suggest discarding conditions as early as possible, and leaving the "bulk" of the logic at the end of the method, as it allows you to mentally track less conditions when reading that logic.

```suggestion
  if (!segment.cfi || !filters.cfi) {
    return true;
  }
  
  const [cfiStart, cfiEnd] = filters.cfi.value.split('-');
  return cfiInRange(segment.cfi, cfiStart, cfiEnd);
```

Reply:
In general I also prefer early-exit. It was written this way because I expect to add other filters in future. In that case the code will look like:

```js
if (segment.cfi && filters.cfi) {
  // Match on CFI
} else if (segment.pageRange && filters.page) {
  // Test if page range intersects
} else {
  return false;
}
```

Reply:
In that case I guess it's ok for now.